lms/static/scripts/frontend_apps/components/SubmitGradeForm.js

```diff
@@ -20,18 +20,46 @@ import ValidationMessage from './ValidationMessage';
 export default function SubmitGradeForm({ disabled = false, student }) {
   const [showError, setShowError] = useState(false); // Is there a validation error message to show?
   const [gradeErrorMessage, setGradeErrorMessage] = useState(''); // The actual validation error message.
+  const [grade, setGrade] = useState('');
```

Comment:
At first blush, it's challenging to differentiate between a `grade` that is a fetched-existing-grade from a `grade` that is entered and submitted. Possibly a naming confusion?

Reply:
sure, grade is always the one fetched, the one submitted is directly from the ref value, but I could clarify that. This should be a read-only grade perhaps.